tests/simulation/test_particle_tracker.py

```diff
@@ -123,6 +116,9 @@ def test_particle_tracker_constructor_errors(
         save_routine = request.getfixturevalue(save_routine)
 
     with pytest.raises(expected_exception):
+        warnings.filterwarnings(
```

Comment:
Most of the changes in this file are just adding this warning filter, which catches the warning thrown when quantities are not zero on the edge of a grid. That is the case because these are small, contrived grids for testing, and we expect it.